VCU/Phantom/Drivers/RTOS/phantom_queue.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * phantom_queue.h
+ *
+ * Created on: July 15, 2022
+ *      Author: Josh Guo
+ * 
+ * Light-weight wrapper for freertos queue API
+ */
+
+#include "phantom_queue.h"
+
+QueueHandle_t Phantom_createQueue(UBaseType_t queue_length, UBaseType_t item_size)
+{
+    QueueHandle_t qHandle = xQueueCreate(queue_length, item_size);
+    while (qHandle == NULL);
+    return qHandle;
+}
+
+QueueSetHandle_t Phantom_createQueueSet(UBaseType_t combined_queue_size)
+{
+    QueueSetHandle_t qSetHandle = xQueueCreateSet(combined_queue_size);
+    while (qSetHandle == NULL);
+    return qSetHandle;
+}
+
```

Comment:
Better to use a message struct that comes with both id and data to know where the data source is from rather than a set